src/test/datascience/widgets/standard.vscode.test.ts

```diff
@@ -73,32 +76,111 @@ suite('Standard IPyWidget (Execution) (slow) (WIDGET_TEST)', function () {
+            'templateFile' in options
+                ? Uri.file(await createTemporaryNotebook(options.templateFile, disposables))
+                : Uri.file(options.notebookFile);
         await openNotebook(nbUri.fsPath);
         await waitForKernelToGetAutoSelected();
-        const comms = initializeWidgetComms(api.serviceContainer);
-        // Confirm we have execution order and output.
-        const cell1 = vscodeNotebook.activeNotebookEditor?.document.cellAt(0)!;
+        return initializeWidgetComms(api.serviceContainer);
+    }
+    async function executionCell(cell: NotebookCell, comms: Utils) {
         await Promise.all([
-            runCell(cell1),
-            waitForExecutionCompletedSuccessfully(cell1),
-            waitForCondition(
-                async () => cell1.outputs.length > 0,
-                defaultNotebookTestTimeout,
-                'Cell output is not empty'
-            ),
+            runCell(cell),
+            waitForExecutionCompletedSuccessfully(cell),
+            waitForCondition(async () => cell.outputs.length > 0, defaultNotebookTestTimeout, 'Cell output is empty'),
             comms.ready
         ]);
+    }
+    async function testSliderWidget(comms: Utils) {
+        // Confirm we have execution order and output.
+        const cell = vscodeNotebook.activeNotebookEditor?.document.cellAt(0)!;
+        await executionCell(cell, comms);
+
         // Verify the slider widget is created.
         await waitForCondition(
             async () => {
-                const innerHTML = await comms.queryHtml('.widget-readout', cell1.outputs[0].id);
+                const innerHTML = await comms.queryHtml('.widget-readout', cell.outputs[0].id);
                 assert.strictEqual(innerHTML, '666', 'Slider not renderer with the right value.');
                 return true;
             },
             WidgetRenderingTimeoutForTests,
             'Slider not rendered'
         );
+    }
+
+    test('Slider Widget', async function () {
+        const comms = await initializeNotebook({ templateFile: templateNbPath });
+        await testSliderWidget(comms);
+    });
+    test('Checkbox Widget', async () => {
+        const comms = await initializeNotebook({ templateFile: templateNbPath });
+        // Confirm we have execution order and output.
+        const cell = vscodeNotebook.activeNotebookEditor?.document.cellAt(2)!;
+        await executionCell(cell, comms);
+
+        await waitForCondition(
+            async () => {
+                const innerHTML = await comms.queryHtml('.widget-checkbox', cell.outputs[0].id);
+                assert.include(innerHTML, 'Check me');
+                assert.include(innerHTML, '<input type="checkbox');
+                return true;
+            },
+            WidgetRenderingTimeoutForTests,
+            'Checkbox not rendered'
+        );
+    });
+    test.skip('Widget renders after executing a notebook which was saved after previous execution', async () => {
```

Comment:
Doesn't work, will need to fix this (possibly something wrong in our code)
Looks like a regression.
Works only if you wait for some time

Reply:
Does the test need to be skipped and an issue filed then? Or were you looking at this fix as part of this PR?

Reply:
Whoops, totally missed the skip that was already there.

Reply:
Seems like a regression? Didn't we have this bug before?
https://github.com/microsoft/vscode-jupyter/issues/8431

Reply:
Aah, yes its a duplicate then.
This wasn't fixed, its actually flaky, it works locally (seems to be 100% of the time).
Must be some race condition (i think thats obvious).

Will re-open #8431
File https://github.com/microsoft/vscode-jupyter/issues/8749 & https://github.com/microsoft/vscode-jupyter/issues/8749

I.e. we have 3 issues related to widgets, but they all seem to work locally (at least for me)
Possible its just flaky

Reply:
Damn it, looks like restart tests are also flaky